### PR TITLE
Add conformance profile for NGINX Gateway Fabric

### PR DESCRIPTION
**Pull request by sjberman**
**What type of PR is this?**
/area conformance

**What this PR does / why we need it**:
Updates conformance profile for NGINX Gateway Fabric.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: conformance/reports/v1.1.0/nginxinc-nginx-gateway-fabric/experimental-1.4.0-default-report.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+date: "2024-08-20T18:43:28Z"
+gatewayAPIChannel: experimental
+gatewayAPIVersion: v1.1.0
+implementation:
+  contact:
+  - https://github.com/nginxinc/nginx-gateway-fabric/discussions/new/choose
+  organization: nginxinc
+  project: nginx-gateway-fabric
+  url: https://github.com/nginxinc/nginx-gateway-fabric
+  version: v1.4.0
```
Comment (robscott) [inline conformance/reports/v1.1.0/nginxinc-nginx-gateway-fabric/experimental-1.4.0-default-report.yaml:11]: Tiny nit: Would be great to have a corresponding update to readme.md and implementations.md

**Review by robscott — COMMENTED** [review]
Thanks @sjberman!

/lgtm
/approve
/ok-to-test

Adding hold if you want to address my comment in this PR, otherwise feel free to remove hold and file a follow up.

/hold

File: conformance/reports/v1.1.0/nginxinc-nginx-gateway-fabric/experimental-1.4.0-default-report.yaml (hunk 1 above)
Comment (sjberman) [reply conformance/reports/v1.1.0/nginxinc-nginx-gateway-fabric/experimental-1.4.0-default-report.yaml:11]: What change are you looking for in implementations.md? I don't see our version listed there.

Comment (robscott) [reply conformance/reports/v1.1.0/nginxinc-nginx-gateway-fabric/experimental-1.4.0-default-report.yaml:11]: Usually link to the latest conformance report from both.

Comment (sjberman) [reply conformance/reports/v1.1.0/nginxinc-nginx-gateway-fabric/experimental-1.4.0-default-report.yaml:11]: Ours just links to the top level nginx conformance directory, not a specific report.

**Comment by robscott** [comment]
Thanks @sjberman!

/lgtm
/hold cancel

**Pull request merged**